x-pack/plugins/index_management/public/application/components/no_match/no_match.tsx

```diff
@@ -6,13 +6,86 @@
  */
 
 import React from 'react';
+import { EuiButton, EuiEmptyPrompt } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
+import { ExtensionsService } from '../../../services';
+import { CreateIndexButton } from '../../sections/home/index_list/create_index/create_index_button';
 
-export const NoMatch = () => (
-  <div data-test-subj="noIndicesMessage">
-    <FormattedMessage
-      id="xpack.idxMgmt.noMatch.noIndicesDescription"
-      defaultMessage="No indices to show"
+export const NoMatch = ({
+  loadIndices,
+  filter,
+  resetFilter,
+  extensionsService,
+}: {
+  loadIndices: () => void;
+  filter: string;
+  resetFilter: () => void;
+  extensionsService: ExtensionsService;
+}) => {
+  if (filter) {
+    return (
+      <EuiEmptyPrompt
+        data-test-subj="noIndicesMessage"
+        title={
+          <h3>
+            <FormattedMessage
+              id="xpack.idxMgmt.noMatch.noIndicesTitle"
+              defaultMessage="No indices found"
+            />
+          </h3>
+        }
+        body={
+          <p>
+            <FormattedMessage
+              id="xpack.idxMgmt.noMatch.noIndicesDescription"
+              defaultMessage="No indices match your search. Clear the searchbar input or click the button below."
+            />
+          </p>
+        }
+        actions={
+          <EuiButton
+            onClick={resetFilter}
+            fill
+            color="primary"
+            iconType="cross"
+            data-test-subj="clearIndicesSearch"
+          >
+            <FormattedMessage
+              id="xpack.idxMgmt.noMatch.clearSearchButton"
+              defaultMessage="Clear search"
+            />
+          </EuiButton>
+        }
+      />
+    );
+  }
+
+  if (extensionsService.emptyListContent) {
+    return extensionsService.emptyListContent.renderContent({
+      createIndexButton: <CreateIndexButton loadIndices={loadIndices} />,
+    });
+  }
+
+  return (
+    <EuiEmptyPrompt
+      data-test-subj="createIndexMessage"
+      title={
+        <h3>
+          <FormattedMessage
+            id="xpack.idxMgmt.noMatch.createIndexTitle"
+            defaultMessage="No indices"
+          />
+        </h3>
+      }
+      body={
+        <p>
+          <FormattedMessage
+            id="xpack.idxMgmt.noMatch.createIndexDescription"
+            defaultMessage="You don't have any indices yet. Create an index by clicking the button below."
```

Comment:
```suggestion
            defaultMessage="Create an index."
```

Reply:
Thanks a lot for the suggestion, @amyjtechwriter! There is a duplication in wording with the "create index" button though, so I'd like to keep "You don't have any indices yet." but will remove the rest of the description to make it shorter.